src/tree/updater_gpu_hist.cu

```diff
@@ -727,13 +728,16 @@ struct DeviceShard {
       dh::safe_cuda(cudaMemsetAsync(left_counts.data().get(), 0,
                                     sizeof(int64_t) * left_counts.size()));
     }
-    thrust::sequence(ridx.CurrentDVec().tbegin(), ridx.CurrentDVec().tend());
+    thrust::sequence(
+        thrust::device_pointer_cast(ridx.CurrentSpan().data()),
```

Comment:
I thought any iterator would do,  why creating a wrapper?

Reply:
If you pass a raw pointer thrust thinks the memory is on the host and crashes.